server/broadcast.go

```diff
@@ -397,7 +397,7 @@ func (bsm *BroadcastSessionsManager) shouldSkipVerification(sessions []*Broadcas
 	if !includesSession(sessions, bsm.verifiedSession) {
 		return false
 	}
-	return common.RandomUintUnder(bsm.VerificationFreq) == 0
+	return common.RandomUintUnder(bsm.VerificationFreq) != 0
```

Comment:
Ooof good catch!

IIUC this method generates a random integer `x` s.t. `0 <= x < VerificationFreq` and there would be a `1 / VerificationFreq` probability for the random integer to be 0. Since this method is checking if verification should be *skipped*, if the check on this line is whether the integer is equal to 0 then this method would return true `1 / VerificationFreq` of the time which is the opposite of what we want i.e. this method should return true `(VericationFreq - 1) / VerificationFreq)` of the time.